### PR TITLE
Support PHP 8.4 property hooks as virtual Eloquent attributes

### PR DESCRIPTION
**Pull request by standaniels**
This PR adds support for treating PHP 8.4+ property hooks as attributes in hasAttribute(). This enables native getter/setter hooks to be recognized during attribute resolution, similar to mutators.

Only applies on PHP >= 8.4 and doesn’t affect existing behavior or lower versions.

I’m not a regular Laravel contributor, so please let me know if I’ve overlooked any edge cases or conventions.

### CODE REVIEW

**Comment by calebdw** [comment]
You've only updated `hasAttribute()`, what about `getAttribute`/`getAttributes`?

File: tests/Database/DatabaseConcernsHasAttributesTest.php [hunk 1]
```diff
@@ -61,6 +62,24 @@ public function testUnsettingCachedAttribute()
...
         $this->assertFalse($instance->cachedAttributeIsset('cacheableProperty'));
     }
+
+    #[RequiresPhp('>= 8.4.0')]
+    public function testCanCastAttributeWithGetPropertyHook()
+    {
+        $class = require __DIR__.'/stubs/EloquentModelWithPropertyHook.php';
+        $class->first_name = 'John';
+        $class->last_name = 'Doe';
+        $this->assertSame('John Doe', $class->full_name);
```
Comment (calebdw) [inline tests/Database/DatabaseConcernsHasAttributesTest.php:72]: This only tests property access (which works because it is an actual property), but you still have the following cases:
- array access (`$class['full_name']`)
- public api (`$class->getAttribute('full_name')`), `$class->getAttributes()` should likely have it as well
- converting model to an array (`$class->toArray()`) should contain `full_class` if it's in the `$appends` property

File: tests/Database/DatabaseConcernsHasAttributesTest.php [hunk 2]
```diff
@@ -61,6 +62,24 @@ public function testUnsettingCachedAttribute()
...
+    }
+
+    #[RequiresPhp('>= 8.4.0')]
+    public function testCanCastAttributeWithSetPropertyHook()
+    {
+        $class = require __DIR__.'/stubs/EloquentModelWithPropertyHook.php';
+        $class->full_name = 'John Doe';
```
Comment (calebdw) [inline tests/Database/DatabaseConcernsHasAttributesTest.php:79]: You need to make sure the following cases also mutate the underlying attributes:
- `new Class(['full_name' => 'John Doe'])`
- `$class->fill(['full_name' => 'John Doe'])`
- `$class['full_name'] = 'John Doe'`

**Review by calebdw — COMMENTED** [review]
I'm interested in the potential here, but this implementation is largely incomplete. Take a look at the Attribute tests for all the edge cases. If this is going to be a replacement for Attributes then this needs to have all of their functionality. This includes:
- acting as an **accessor** (changes database column on read)
- acting as a **mutator** (updates other column(s) on write)
- acting as a **computed property** (a readonly property that does not exist in the database)
  - computed properties should also support being added to the `$appends` array

**Comment by taylorotwell** [comment]
Hey there - see comments above. Going to close for now.

File: tests/Database/DatabaseConcernsHasAttributesTest.php (hunk 1 above)
Comment (standaniels) [reply tests/Database/DatabaseConcernsHasAttributesTest.php:72]: The new commits are in [standaniels/laravel-framework:property-hooks](https://github.com/standaniels/laravel-framework/tree/property-hooks) but somehow this PR isn't updated.

Comment (calebdw) [reply tests/Database/DatabaseConcernsHasAttributesTest.php:72]: You'll need to open a new PR
